channel/tests/test_outgoing.py

```diff
@@ -0,0 +1,404 @@
+    mock_get_bot_by_session_id = AsyncMock(
+        return_value=("test_number", "encrypted_credentials")
+    )
+    mock_decrypt_credentials = MagicMock(return_value={"whatsapp": "api_key"})
+    mock_wa_send_form = MagicMock(return_value="test_channel_id")
+    mock_create_message = AsyncMock()
+
+    with patch(
+        "src.handlers.outgoing.get_user_by_session_id", mock_get_user_by_session_id
+    ):
+        with patch(
+            "src.handlers.outgoing.get_bot_by_session_id", mock_get_bot_by_session_id
+        ):
+            with patch(
+                "src.handlers.outgoing.decrypt_credentials", mock_decrypt_credentials
+            ):
+                with patch(
+                    "lib.whatsapp.WhatsappHelper.wa_send_form", mock_wa_send_form
+                ):
+                    with patch(
+                        "src.handlers.outgoing.create_message", mock_create_message
+                    ):
+                        message = ChannelInput(
+                            source="language",
+                            message_id="test_msg_id",
+                            turn_id="test_turn_id",
+                            session_id="test_session_id",
+                            intent=ChannelIntent.BOT_OUT,
+                            data=BotOutput(
+                                message_data=MessageData(message_text="Form body"),
+                                message_type=MessageType.FORM,
+                                footer="Form footer",
+                                wa_flow_id="flow_id",
+                                wa_screen_id="screen_id",
+                                form_token="form_token",
+                            ),
+                            dialog="",
+                        )
+                        await send_message_to_user(message)
+                        mock_wa_send_form.assert_called_once_with(
+                            wa_bnumber="test_number",
+                            wa_api_key="api_key",
+                            user_tele="1234567890",
+                            flow_id="flow_id",
+                            screen_id="screen_id",
+                            body="Form body",
+                            footer="Form footer",
+                            token="form_token",
+                        )
+                        mock_create_message.assert_called_once()
+
+
+@pytest.mark.asyncio
+async def test_send_language_message():
+    mock_get_user_by_session_id = AsyncMock(
+        return_value=MagicMock(phone_number="1234567890")
+    )
+    mock_get_bot_by_session_id = AsyncMock(
+        return_value=("test_number", "encrypted_credentials")
+    )
+    mock_decrypt_credentials = MagicMock(return_value={"whatsapp": "api_key"})
+    mock_wa_send_text_message = MagicMock(return_value="test_channel_id")
+    mock_wa_send_interactive_message = MagicMock(return_value="test_channel_id")
+    mock_create_message = AsyncMock()
+
+    with patch(
```

Comment:
Is there a better way to write this?

with 
   with
      with
         ...